packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js

```diff
@@ -47,21 +47,77 @@ function SingleImageViewer({
   zoomControlFn = null,
   zooming = true
 }) {
+  const styledSVGRef = useRef(null)
+  const [svgDimensions, setSvgDimensions] = useState({ width: '100%', height: '100%' })
+  
+  const [showZoomHelper, setShowZoomHelper] = useState(false)
+  const [fadingOut, setFadingOut] = useState(false)
+
   useEffect(function onMount() {
     enableRotation()
   }, [])
 
+  useEffect(function onDimensionChange() {
+    // Update dimensions after component has mounted and whenever natural width/height changes
+    if (styledSVGRef.current) {
+      function updateDimensions() {
+        setSvgDimensions({
+          width: `${styledSVGRef.current.clientWidth}px`,
+          height: `${styledSVGRef.current.clientHeight}px`
+        })
+      }
+      
+      updateDimensions()
+    }
+  }, [naturalWidth, naturalHeight, styledSVGRef])
+
+  // Handle the first scroll event
+  function handleFirstScroll() {
+    if (allowsScrolling && zooming) {
+      setShowZoomHelper(true)
+      
+      // Set fading out after 3 seconds
+      setTimeout(() => {
+        setFadingOut(true)
+      }, 3000)
+      
+      // Hide completely after animation completes (300ms animation)
+      setTimeout(() => {
+        setShowZoomHelper(false)
+        setFadingOut(false)
+      }, 3300)
+    }
+  }
 
   const maxHeight = limitSubjectHeight ? `min(${naturalHeight}px, 90vh)` : null
   const maxWidth = limitSubjectHeight ? `${naturalWidth}px` : '100%'
 
+  // If the image is landscape, set the top margin to 60px
+  // If the image is portrait, set the top margin to 240px (ImageToolbar with fewest options height of 360px - ZoomHelperOverlay message box height of 120px)
+  const zoomHelperMessageTop = naturalHeight > naturalWidth ? '240px' : '60px'
+
   return (
     <>
       {zoomControlFn && (
         <ZoomControlButton onClick={zoomControlFn} zooming={zooming} />
       )}
-      <Box align="flex-end" animation="fadeIn" overflow="hidden" width="100%">
+      <Box
+        align='flex-end'
+        animation='fadeIn'
+        overflow='hidden'
+        width='100%'
+        position='relative'
```

Comment:
`position` is not recognized by Grommet Box, so you'll need to make this a `styled` component to use `relative`. You can see in the UI `relative` is not working because the top of the overlay is aligned with the banner in this i-fancy-cats screenshot, and it does not stretch to fill the height of the subject:
<img width="545" alt="Screenshot 2025-04-02 at 12 42 01 PM" src="https://github.com/user-attachments/assets/4c10d638-b25b-4070-a727-3da8eeab9657" />

Reply:
Thanks, great catch! Do you have a preference if the other styling props stay with the component in the `render` or move to the styled-component?